### PR TITLE
Allow assume role option, centralize aws client config

### PR DESCRIPTION
**Pull request by krjackso**

### CODE REVIEW

File: bin/cumulus.rb [hunk 1]
```diff
@@ -622,9 +627,10 @@ def help_message
 
 puts "Reading configuration from '#{options[:config]}'" if options[:verbose]
 
-Cumulus::Configuration.init(options[:config], options[:profile], options[:autoscaling_force_size])
+Cumulus::Configuration.init(options[:config], options[:profile], options[:assume_role], options[:autoscaling_force_size])
 
 puts "Using aws profile '#{options[:profile]}'" if options[:verbose]
+puts "Using assume role '#{options[:assume_role]}'" if options[:verbose]
```
Comment (msiebert) [inline bin/cumulus.rb:633]: Would `"Assuming role '#..'"` be more accurate?

File: lib/conf/Configuration.rb [hunk 2]
```diff
@@ -107,6 +107,21 @@ def initialize(conf_dir, profile, autoscaling_force_size)
...
+          role_arn: assume_role,
+          role_session_name: "#{region}-#{@profile}"
+        )
+      end
+
+      @client = {
+        :region => region,
+        :profile => profile,
+        :credentials => credentials,
+      }.reject { |_, v| v.nil? }
```
Comment (msiebert) [inline lib/conf/Configuration.rb:124]: if only we were on that higher version of ruby...

**Comment by msiebert** [comment]
Looks good to me. This is some nice stuff.

File: bin/cumulus.rb (hunk 1 above)
Comment (krjackso) [inline bin/cumulus.rb:633]: Maybe, the CLI config uses "assume_role = arn" which I had thought would be pulled in when using that profile.

**Pull request merged**